src/routers/v2/DelaysRouter.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import ApplicationRouter from '../../appdev/ApplicationRouter';
+import RealtimeFeedUtils from '../../utils/RealtimeFeedUtils';
+
+class DelaysRouter extends ApplicationRouter<Object> {
+  constructor() {
+    super(['POST']);
+  }
+
+  getPath(): string {
+    return '/delays/';
+  }
+
+  async content(req): Promise<any> {
+    const delays = req.body.data;
+    const rtf = await RealtimeFeedUtils.fetchRTF();
+    const delaysPromises = delays.map(async (stopAndTripID) => {
+      const tripID = await stopAndTripID.tripID;
+      const stopID = await stopAndTripID.stopID;
+      const res = await RealtimeFeedUtils.getDelayInformation(stopID, tripID, rtf);
+      if (res) {
+        stopAndTripID.delay = res.delay;
+      } else {
+        stopAndTripID.delay = null;
+      }
+    });
+    await Promise.all(delaysPromises);
+    return delays;
```

Comment:
This is confusing! `delays` is a `const`, which implicitly means that it shouldn't be modified, so someone quickly reading this code would assume that we're incorrectly returning a value that hasn't been modified since we set it.

Why not `let` assign `delays` above, and not create a new variable for `delaysPromises`? You can also return `Promise.all(delaysPromises)`, since this function itself is also `async`.

Reply:
the linter actually prefers const because i'm technically not modifying it?
<img width="616" alt="Screen Shot 2019-10-06 at 1 55 18 PM" src="https://user-images.githubusercontent.com/13739525/66273386-66662800-e841-11e9-9b6c-5121e7ae3590.png">

Reply:
Is this error because of the `const`? Isn't it because you're not destructuring, as @meganle noted?